### PR TITLE
Pod monitoring

### PR DESCRIPTION
**Pull request by copejon**
Address #83 

Exploratory PR to share work on adding pod monitoring to the controller.

### CODE REVIEW

**Comment by copejon** [comment]
Pods aren't being added to the pod queue for some reason.

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -61,43 +81,91 @@ func NewController(client kubernetes.Interface, pvcInformer cache.SharedIndexInf
 }
 
 func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
-	defer c.queue.ShutDown()
+	defer c.pvcQueue.ShutDown()
```
Comment (jeffvance) [inline pkg/controller/controller.go]: don't need to also shutdown the podQueue?

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -61,43 +81,91 @@ func NewController(client kubernetes.Interface, pvcInformer cache.SharedIndexInf
...
+	glog.Infof("[=== DEBUG ===] getting pod from store by key %q\n", keyString)
+	obj, ok, err := c.podInformer.GetIndexer().GetByKey(keyString)
+	if err != nil || !ok {
+		return false
 	}
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		return false
+	}
+	if err := c.processPodItem(pod); err == nil {
```
Comment (jeffvance) [inline pkg/controller/controller.go]: I thought we want to Forget for success case and not forget if there was an error?

Comment (copejon) [reply pkg/controller/controller.go]: right, err is nil here, so the process succeeded.  Forget the key so we don't keep requeueing it

File: pkg/controller/controller.go (hunk 1 above)
Comment (copejon) [reply pkg/controller/controller.go]: yep! will fix, thanks.

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -109,7 +177,7 @@ func (c *Controller) ProcessNextItem() bool {
 	glog.Infof("processNextItem: next pvc to process: %s\n", key)
 
 	// all checks have passed, let's process it!
-	if err := c.processItem(pvc); err != nil {
+	if err := c.processPvcItem(pvc); err != nil {
```
Comment (jeffvance) [inline pkg/controller/controller.go]: this should be `== nil`, right?

Comment (copejon) [reply pkg/controller/controller.go]: yes.  That's still in master, I'll open a small fix-up pr to change it

**Comment by copejon** [comment]
Moved branch to kubevirt/containerized-data-importer, opening PR from there

**Pull request closed without merging**